FreeSurfer/LongitudinalFreeSurferPipeline.sh

```diff
@@ -0,0 +1,418 @@
+                #copy anything unrecognized
+                newargs+=("${origargs[i]}")
+                ;;
+        esac
+    done
+    if ((changeargs))
+    then
+        echo "original arguments: $*"
+        set -- "${newargs[@]}"
+        echo "new arguments: $*"
+        echo "extra recon-all base arguments: ${extra_reconall_args_base_manual[*]+"${extra_reconall_args_base_manual[*]}"}"
+        echo "extra recon-all long arguments: ${extra_reconall_args_long_manual[*]+"${extra_reconall_args_long_manual[*]}"}"
+    fi
+fi
+
+#description to use in usage - syntax of parameters is now explained automatically
+opts_SetScriptDescription "Runs the Longitudinal FreeSurfer HCP pipeline"
+
+# Show usage information
+opts_AddMandatory '--subject' 'SubjectID' 'subject' "Subject ID (required)  Used with --path input to create full path to root directory for all outputs generated as path/subject"
+
+opts_AddMandatory '--path' 'StudyFolder' 'path' "Path to subject's data folder (required)  Used with --subject input to create full path to root directory for all outputs generated as path/subject)"
+
+opts_AddMandatory '--sessions' 'Sessions' 'sessions' "An @ symbol separated list of session IDs (required)"
+
+opts_AddMandatory '--template-id' 'TemplateID' 'template-id' "An @ symbol separated list of session IDs (required)"
+
+opts_AddOptional '--seed' 'recon_all_seed' "Seed" 'recon-all seed value'
+
+#TSC: repeatable options aren't currently supported in newopts, do them manually and fake the help info for now
+opts_AddOptional '--extra-reconall-arg-base' 'extra_reconall_args_base' 'token' "(repeatable)  Generic single token argument to pass to recon-all for base template preparation.  Provides a mechanism to:  (i) customize the recon-all command  (ii) specify the recon-all stage(s) to be run (e.g., in the case of FreeSurfer edits)  If you want to avoid running all the stages inherent to the '-all' flag in recon-all,  you also need to include the --existing-subject flag.  The token itself may include dashes and equal signs (although Freesurfer doesn't currently use  equal signs in its argument specification).  e.g., --extra-reconall-arg-base=-3T is the correct syntax for adding the stand-alone '-3T' flag to recon-all.  But, --extra-reconall-arg-base='-norm3diters 3' is NOT acceptable.  For recon-all flags that themselves require an argument, you can handle that by specifying  --extra-reconall-arg-base multiple times (in the proper sequential fashion).  e.g., --extra-reconall-arg-base=-norm3diters --extra-reconall-arg-base=3  will be translated to '-norm3diters 3' when passed to recon-all."
+
+opts_AddOptional '--generate_template_only' 'generate_template_only' 'integer' 'Set to 1 to generate template only without timepoints [0]'
+opts_AddOptional '--generate_timepoints_only' 'generate_timepoints_only' 'integer' 'Set to 1 to generate timepoints only (assuming the template has been generated previously with "--generate_template_only" option) [0]'
```

Comment:
When unspecified like this, the default is actually the empty string.  Bash interprets (("")) to be false, so it sort of works, but not for obvious reasons.

As mentioned in the PR discussion, I would suggest reworking the logic (and not use underscores in the flags) to:
```
opts_AddOptional '--generate-template' 'generateTemplateStr' 'TRUE or FALSE' "generate the cross-timepoint template output, default TRUE" 'TRUE'
opts_AddOptional '--generate-timpoints' 'generateTimepointsStr' 'TRUE or FALSE' "generate the per-timepoint outputs, default TRUE" 'TRUE'
...
generateTemplate=$(opts_StringToBool "$generateTemplateStr")
generateTimepoints=$(opts_StringToBool "$generateTimepointsStr")
```